ipatests/test_integration/test_installation.py

```diff
@@ -567,3 +567,51 @@ def test_KRA_install_after_cert_renew(self):
         self.master.run_command(['kinit', 'admin'], stdin_text=passwd)
         cmd = self.master.run_command(['ipa-kra-install', '-p', dm_pass, '-U'])
         self.master.run_command(['systemctl', 'start', 'chronyd'])
+
+
+class TestMaskMasterInstall(IntegrationTest):
+    """ Test master installation with wrong mask
+
+    This test checks that master installation fails (expectedly) if
+    mask > 022.
+
+    related ticket: https://pagure.io/freeipa/issue/7193
+    """
+
+    num_replicas = 0
+
+    @classmethod
+    def install(cls, mh):
+        pass
+
+    def test_install_master(self):
+        self.master.run_command(['umask', '0027'])
+        cmd = tasks.install_master(self.master)
+        exp_str = ("Unsupported system umask value")
+        assert (exp_str in cmd.stderr_text and cmd.returncode != 0)
+
+
+class TestUmaskReplicaInstall(IntegrationTest):
+    """ Test replica installation with wrong mask
+
+    This test checks that replica installation fails (expectedly) if
+    mask > 022.
+
+    related ticket: https://pagure.io/freeipa/issue/7193
+    """
+
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        pass
+
+    def test_install_master(self):
+        tasks.install_master(self.master, setup_dns=False)
+
+    def test_install_replica(self):
+        self.replicas[0].run_command(['umask', '0027'])
```

Comment:
Same comment as above, the test needs to write the umask in /root/.bashrc so that it gets permanent and does not get lost as soon as `run_command` returns.

Reply:
Thanks for the input! Change is done.